Components/scripts/MapBase.gd

```diff
@@ -281,8 +284,58 @@ func wall(path):
 		changed_tiles[point + up] = true
 		changed_tiles[point + upup] = true
 		changed_tiles[point + downdown] = true
+	
+	return alreadyExistingWalls
+
+func possibleDoors(path):
+	var possibleDoorsInWalls = []
+	var possibleDoors = []
+	
+	if path.size() > 0:
+		var previousMovingHorizontal
+		
+		if (!is_corner_wall(path[0])):
+			possibleDoors.append(Vector2(path[0].x, path[0].y))
+		
+		if path.size() > 1:
+			for index in range(1, path.size()):
+				var point_a = path[index]
+				var point_b = path[index - 1]
+				var diff = point_b - point_a
+				var nextMoveHorizontal
+				
+				if diff[0] == 0 and diff[1] != 0:
+					nextMoveHorizontal = false
+				elif diff[0] != 0 and diff[1] == 0:
+					nextMoveHorizontal = true
+				else:
+					possibleDoorsInWalls.append(possibleDoors)
+					possibleDoors = []
+				
+				if previousMovingHorizontal == null && hasWallMoveChangedDirection(previousMovingHorizontal, nextMoveHorizontal) && possibleDoors.size() > 0:
+					possibleDoorsInWalls.append(possibleDoors)
+					possibleDoors = []
+				
+				previousMovingHorizontal = nextMoveHorizontal
+				
+				if (!is_corner_wall(point_a)):
+					possibleDoors.append(Vector2(point_a.x, point_a.y))
+	
+	possibleDoorsInWalls.append(possibleDoors)
+	
+	return possibleDoorsInWalls;
+
+func hasWallMoveChangedDirection(lastMoveHorizontal, nextMoveHorizontal):
+	return lastMoveHorizontal != nextMoveHorizontal
+
+func is_corner_wall(point):
+	var isVertical = is_door_or_wall(point + Vector2(0, 1)) || is_door_or_wall(point + Vector2(0, -1))
+	var isHorizontal = is_door_or_wall(point + Vector2(1, 0)) || is_door_or_wall(point + Vector2(-1, 0))
+	
+	return isVertical && isHorizontal
```

Comment:
If walls are next to each other, I think it designates every wall as a corner wall. E.g. in the following, # is a wall and _ is space:
# # # # #
# # # # #
_ _ _ _ _

Wouldn't all of the first and second row be designated as corner walls?

This might not be possible to solve, and if it works then it might not be worth changing. I.e. if this is to stop a door being placed on a corner wall, we don't want doors here anyway as they wouldn't be able to get through, so no biggie :)

Reply:
yeah they probably would be, but that won't matter for this story atleat as it is only checked on shared walls